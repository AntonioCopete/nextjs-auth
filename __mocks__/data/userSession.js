export const userSessionMock = {
	name: 'Username',
	email: 'email@domain.com',
	image: 'https://image-url',
};
