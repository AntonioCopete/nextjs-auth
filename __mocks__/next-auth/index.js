const getServerSessionMock = jest.fn();

export const getServerSession = getServerSessionMock;
