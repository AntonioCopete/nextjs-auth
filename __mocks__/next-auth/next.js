const NextAuthMock = jest.fn();

const NextAuth = NextAuthMock;
export default NextAuth;
