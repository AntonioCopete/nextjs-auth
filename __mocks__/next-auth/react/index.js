const signInMock = jest.fn();
const SessionProviderMock = jest.fn();

export const signIn = signInMock;
export const SessionProvider = SessionProviderMock;
