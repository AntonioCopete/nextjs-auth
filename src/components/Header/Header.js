import Link from "next/link";
import styles from "./Header.module.scss";
import { ROUTES } from "../../../constants";

const Header = () => {
  return (
    <header className={styles.header}>
      <Link href={ROUTES.LOGIN}>
        <div className={styles.menuItem}>Login</div>
      </Link>
      <Link href={ROUTES.REGISTER}>
        <div className={styles.menuItem}>Register</div>
      </Link>
    </header>
  );
};

export default Header;
