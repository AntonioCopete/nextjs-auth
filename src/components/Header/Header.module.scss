.header {
  background-color: rgb(185, 205, 255);
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 1.5rem 1rem;
  font-size: 2rem;
  gap: 1rem;
}
