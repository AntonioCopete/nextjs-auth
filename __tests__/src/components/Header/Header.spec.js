import { render } from '@testing-library/react';
import Header from '../../../../src/components/Header/Header';
import { getServerSession } from 'next-auth';
import { userSessionMock } from '../../../../__mocks__/data/userSession';

describe('Header', () => {
	it('should match snapshot when user is logged in', async () => {
		getServerSession.mockReturnValue(userSessionMock);

		const component = render(await Header());
		expect(component).toMatchSnapshot();
	});

	it('should match snapshot when user is NOT logged in', async () => {
		getServerSession.mockReturnValue(null);

		const component = render(await Header());
		expect(component).toMatchSnapshot();
	});

	it('should fail on getServerSession', async () => {
		getServerSession.mockRejectedValue(new Error('getServerSession error'));

		const component = render(await Header());
		expect(component).toMatchSnapshot();
	});
});
