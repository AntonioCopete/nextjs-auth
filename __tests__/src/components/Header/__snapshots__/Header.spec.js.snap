// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should fail on getServerSession 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <a
          href="/login"
        >
          <div
            class="menuItem"
          >
            Login
          </div>
        </a>
        <a
          href="/register"
        >
          <div
            class="menuItem"
          >
            Register
          </div>
        </a>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <a
        href="/login"
      >
        <div
          class="menuItem"
        >
          Login
        </div>
      </a>
      <a
        href="/register"
      >
        <div
          class="menuItem"
        >
          Register
        </div>
      </a>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header should match snapshot when user is NOT logged in 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <a
          href="/login"
        >
          <div
            class="menuItem"
          >
            Login
          </div>
        </a>
        <a
          href="/register"
        >
          <div
            class="menuItem"
          >
            Register
          </div>
        </a>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <a
        href="/login"
      >
        <div
          class="menuItem"
        >
          Login
        </div>
      </a>
      <a
        href="/register"
      >
        <div
          class="menuItem"
        >
          Register
        </div>
      </a>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header should match snapshot when user is logged in 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <a
          href="/login"
        >
          <div
            class="menuItem"
          >
            Login
          </div>
        </a>
        <a
          href="/register"
        >
          <div
            class="menuItem"
          >
            Register
          </div>
        </a>
        <div
          class="menuItemLogout"
        >
          Logout
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <a
        href="/login"
      >
        <div
          class="menuItem"
        >
          Login
        </div>
      </a>
      <a
        href="/register"
      >
        <div
          class="menuItem"
        >
          Register
        </div>
      </a>
      <div
        class="menuItemLogout"
      >
        Logout
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
